samples-generator/bin/3d-tiles-samples-generator.js

```diff
@@ -926,24 +950,59 @@ function saveInstancedTileset(tilesetName, tileOptions, tilesetOptions) {
 
 function saveBatchedTileset(tilesetName, tileOptions, tilesetOptions) {
     var tilesetDirectory = path.join(outputDirectory, 'Batched', tilesetName);
-    var contentUri = lowercase(tilesetName) + '.b3dm';
-    var tilePath = path.join(tilesetDirectory, contentUri);
-    var tilesetPath = path.join(tilesetDirectory, 'tileset.json');
 
     tileOptions = defaultValue(tileOptions, {});
     tileOptions.buildingOptions = defaultValue(tileOptions.buildingOptions, buildingTemplate);
     tileOptions.transform = defaultValue(tileOptions.transform, buildingsTransform);
     tileOptions.relativeToCenter = defaultValue(tileOptions.relativeToCenter, true);
-
     tilesetOptions = defaultValue(tilesetOptions, {});
+
+    var ext = '';
+    if (argv['3d-tiles-next']) {
+        tileOptions.use3dTilesNext = true;
+        tileOptions.useGlb = argv.glb;
+        tileOptions.versionNumber = 1.1;
+        ext = (argv.glb) ? '.glb' : '.gltf';
+    } else {
+        ext = '.b3dm';
+    }
+
+    var contentUri = lowercase(tilesetName) + ext;
     tilesetOptions.contentUri = contentUri;
     tilesetOptions.geometricError = smallGeometricError;
     if (!defined(tilesetOptions.region) && !defined(tilesetOptions.sphere) && !defined(tilesetOptions.box)) {
         tilesetOptions.region = smallRegion;
     }
 
+    var tilePath = path.join(tilesetDirectory, contentUri);
+    var tilesetPath = path.join(tilesetDirectory, 'tileset.json');
+
     return createBuildingsTile(tileOptions)
         .then(function(result) {
+            if (argv['3d-tiles-next']) {
+                tilesetOptions.versionNumber = 1.1;
+
+                if (argv.glb) {
+                    if (tilesetOptions.contentDataUri) {
+                        tilesetOptions.contentUri = 'data:model/gltf-binary;base64,' + Buffer.from(result).toString('base64');
+                    }
+                    return Promise.all([
+                        saveBinary(tilePath, result, tilesetOptions.gzip),
+                        saveJson(tilesetPath, createTilesetJsonSingle(tilesetOptions), prettyJson)
+                    ]);
+                }
+
+                if (tilesetOptions.contentDataUri) {
+                    tilesetOptions.contentUri = 'data:model/gltf+json;base64,' + Buffer.from(JSON.stringify(result)).toString('base64');
+                }
+
+                return Promise.all([
+                    saveJson(tilesetPath, createTilesetJsonSingle(tilesetOptions), prettyJson),
+                    saveJson(tilePath, result, prettyJson)
+                ]);
+            }
+
+            // old .b3dm
             var b3dm = result.b3dm;
             var batchTableJson = result.batchTableJson;
             tilesetOptions.properties = getProperties(batchTableJson);
```

Comment:
Did you see this comment: https://github.com/CesiumGS/3d-tiles-validator/pull/186/files#r386655831?